sources/dataset.py

```diff
@@ -163,6 +139,73 @@ class (useful for semantic segmentation result printing)
                                      "aggregate": aggregate,
                                      "color": color}
 
+    def save(self, filename):
+        """Save dataset in a json file indicated by `filename`
+
+        Parameter
+        ---------
+        filename: object
+            String designing the relative path where the dataset must be saved
+        """
+        with open(filename, 'w') as fp:
+            json.dump({"image_size": self.image_size,
+                       "classes": self.class_info,
+                       "images": self.image_info}, fp)
+        utils.logger.info("The dataset has been saved into {}".format(filename))
+
+    def load(self, filename, nb_images=None):
+        """Load a dataset from a json file indicated by `filename`
+
+        Parameter
+        ---------
+        filename: object
+            String designing the relative path from where the dataset must be
+        loaded
+        nb_images: integer
+            Number of images that must be loaded (if None, the whole dataset is loaded)
+        """
+        with open(filename) as fp:
+            ds = json.load(fp)
+            self.image_size = ds["image_size"]
+            self.class_info = {int(k):ds["classes"][k] for k in ds["classes"].keys()}
+            if nb_images is None:
+                self.image_info = {int(k):ds["images"][k] for k in ds["images"].keys()}
+            else:
+                self.image_info = {int(k):ds["images"][k] for k in ds["images"].keys() if int(k) < nb_images}
```

Comment:
The `nb_images` threshold/filter works only if the keys of the images dataset can be converted to an integer, begin to 0 and are sorted. The `keys()` or dict iteration cannot return a sorted list of keys (it's not warranty by the dict implementation). For instance, `.keys()` could return `["10", "0", "3", "100"]`.

If we change the keys of the dataset, e.g. random letter and number, this won't work.

Reply:
Maybe `enumerate` can do the job

```python
{k: v for idx, (k, v) in enumerate(ds["images"].items()) if idx < nb_images}
```

Reply:
Done.